lib/model/model.go

```diff
@@ -770,12 +798,21 @@ func (m *Model) folderSharedWithLocked(folder string, deviceID protocol.DeviceID
 	return false
 }
 
+func (m *Model) ThrowUnexpectedFolderError (folder protocol.Folder, deviceID protocol.DeviceID) {
```

Comment:
Nitpick, but we don't really talk about throwing in Go and this could be an unexported method, so `unexpectedFolderError` would be OK. If it needs to exist, given that I think we should only need to do this processing in one place.